### PR TITLE
Fix issue #8680 crash with multiple windows with blinds in the same zone

### LINKED ISSUE #8680: Segmentation fault in CalcWinTransDifSolInitialDistribution
Issue overview
--------------

One of the OpenStudio-resources regression test that has ShadingControl in it is segfaulting with 9.5.0. It was working with 9.4.0 and 9.5.0-IOFreeze.

Crash happens here, Lay = 0:

https://github.com/NREL/EnergyPlus/blob/de239b2e5f504e381ef4f296893ef93ce5410215/src/EnergyPlus/SolarShading.cc#L11949

Doing two runs I see two outcomes:

```
/usr/local/EnergyPlus-9-5-0/energyplus -w /home/julien/Software/Others/EnergyPlus/weather/USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw -d out-950 shadingcontrol_singlezone_rb.idf 
```

### Outcome one: converge issue, graceful exit

```
Initializing HVAC
Warming up
**FATAL:Program halted because of convergence error in SolveForWindowTemperatures for window SUB SURFACE 7
EnergyPlus Run Time=00hr 00min  0.43sec
Program terminated: EnergyPlus Terminated--Error(s) Detected.
```

### Outcome two: converge issue doesn't happen, segfault later

```
Initializing HVAC
Warming up
Warming up
Warming up
Performing Zone Sizing Simulation
...for Sizing Period: #1 CHICAGO OHARE INTL AP ANN CLG .4% CONDNS DB=>MWB
Warming up
Warming up
Performing Zone Sizing Simulation
...for Sizing Period: #2 CHICAGO OHARE INTL AP ANN HTG 99.6% CONDNS DB
Adjusting Air System Sizing
Adjusting Standard 62.1 Ventilation Sizing
Initializing Simulation
Reporting Surfaces
Beginning Primary Simulation
Initializing New Environment Parameters
Warming up {1}
Segmentation fault (core dumped)
```

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): all
 - Version of EnergyPlus (if using an intermediate build, include SHA): 9.5.0 official.
 - OS resources test: https://github.com/NREL/OpenStudio-resources/blob/develop/model/simulationtests/shadingcontrol_singlezone.rb

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here): https://github.com/NREL/EnergyPlusDevSupport/blob/3e9a93cb9110859cda81ebfb006e48ab50c8ac55/DefectFiles/8000s/8680/shadingcontrol_singlezone_rb.idf#L1
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by xuanluo113**
Pull request overview
---------------------
 - Fixes #8680 
 - 
Fixed a typo.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/SolarShading.cc [hunk 1]
```diff
@@ -11946,7 +11946,7 @@ void CalcWinTransDifSolInitialDistribution(EnergyPlusData &state)
                                             construct_sh_BlAbsDiffBack(std::min(MaxSlatAngs, SurfWinSlatsAngIndex + 1), IGlass),
                                             SurfWinSlatsAngInterpFac);
                                     } else {
-                                        BlAbsDiffBk = construct_sh_BlAbsDiffBack(1, Lay);
+                                        BlAbsDiffBk = construct_sh_BlAbsDiffBack(1, IGlass);
```
Comment (jmarrec) [inline src/EnergyPlus/SolarShading.cc:11949]: Original regression from https://github.com/NREL/EnergyPlus/commit/57aa34f2d6cd0cba59df94b5e49ddc98f9031bda#diff-18fbb8a5b1334b62513de2441bdb4bb4ef1fbe5fbec24d5dd0effcd019dfc33dR10986

I think It should be IGlass indeed, fix looks good to me :+1:

**Review by jmarrec — APPROVED** [review]

**Comment by xuanluo113** [comment]
@jmarrec Thanks for the review. This branch is still hanging. Is this coverage issue blocking the merge?

**Comment by mjwitte** [comment]
@xuanluo113 I agree that the fix is correct, but I'm concerned that this was not caught by the CI regression/integration tests. What feature in the defect file trips on this defect? We should consider adding a unit test or modifying an integration test file to cover this code. Also, does the integration and unit test code coverage show any other blocks of code in the solars hading functions that are not covered?

**Comment by xuanluo113** [comment]
@mjwitte It seems to need two windows in the same enclosure on different walls (with a view factor for each other), and at least one of them has blinds to trigger this branch.
The [OS test file](https://github.com/NREL/EnergyPlusDevSupport/blob/3e9a93cb9110859cda81ebfb006e48ab50c8ac55/DefectFiles/8000s/8680/shadingcontrol_singlezone_rb.idf#L1) @jmarrec provided includes this case with `Sub Surface 4` and `Sub Surface 4` in the same zone, and both has interior blinds.

I can modify the current `PurchAirWindowBlind.idf` to add another exterior window on the west wall of the West zone (attached) and hopefully would solve this. Any suggestions?

Screen Shot 2021-04-07 at 12 10 07 PM
Screen Shot 2021-04-07 at 11 15 57 AM

**Comment by mjwitte** [comment]
@xuanluo113 That would be good. Please make a separate PR for that testfile change, to confirm that CI reports a failure.

**Comment by xuanluo113** [comment]
@mjwitte Thanks. Updated in #8702.

**Review by mjwitte — APPROVED** [review]
Tested this branch with the modified PurhAirWindowBlind (from #8702 ) and compared results with v9.4.0. No diffs (except in thermal comfort outputs which are expected from 9.4 to 9.5). This will merge once CI comes back green.

**Comment by mjwitte** [comment]
Clean and green. Merging.

**Pull request merged**